src/main/java/org/veupathdb/service/eda/ds/plugin/pass/PieplotPlugin.java

```diff
@@ -0,0 +1,97 @@
+package org.veupathdb.service.eda.ds.plugin.pass;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import org.gusdb.fgputil.ListBuilder;
+import org.gusdb.fgputil.validation.ValidationException;
+import org.json.JSONObject;
+import org.veupathdb.service.eda.common.client.spec.StreamSpec;
+import org.veupathdb.service.eda.ds.constraints.ConstraintSpec;
+import org.veupathdb.service.eda.ds.constraints.DataElementSet;
+import org.veupathdb.service.eda.ds.plugin.AbstractPlugin;
+import org.veupathdb.service.eda.generated.model.PieplotPostRequest;
+import org.veupathdb.service.eda.generated.model.PieplotSpec;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import static org.veupathdb.service.eda.ds.metadata.AppsMetadata.CLINEPI_PROJECT;
+import static org.veupathdb.service.eda.ds.metadata.AppsMetadata.MICROBIOME_PROJECT;
+import static org.veupathdb.service.eda.ds.util.RServeClient.streamResult;
+import static org.veupathdb.service.eda.ds.util.RServeClient.useRConnectionWithRemoteFiles;
+
+public class PieplotPlugin extends AbstractPlugin<PieplotPostRequest, PieplotSpec> {
+
+  @Override
+  public String getDisplayName() {
+    return "Pie plot";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Visualize part-to-whole relationships across the categories of a variable";
+  }
+  
+  @Override
+  protected Class<PieplotSpec> getVisualizationSpecClass() {
+    return PieplotSpec.class;
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
+    return new ConstraintSpec()
+      .dependencyOrder("overlayVariable", "facetVariable")
+      .pattern()
+        .element("overlayVariable")
+          .description("Variables with more than 8 values will assign the top 7 values by count to their own categories and assign the additonal values into an 'other' category.")
```

Comment:
Just a note - we should also describe this elsewhere, maybe in the map docs. If i remember correctly i think this description only shows if the variable is disabled, which would not be the case for us